app/models/user.rb

```diff
@@ -1,11 +1,16 @@
 class User < ApplicationRecord
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  before_save :downcase_email
 
-  validates :name, :password, :confirm_password, presence: true
```

Comment:
Vì sao không cần `:password, :confirm_password` presence: true nữa em nhỉ

Reply:
- validates min length 10 cho password  sẽ cover luôn presence: true (not nil, not empty string)
- confirm_password: ở đây không validates vì has_secure_password sẽ validates (có khớp giữa password và password_confirm)
- has_secure_password sẽ không trigger validates nếu password_confirm là nil